onnx/numpy_helper.py

```diff
@@ -221,6 +221,83 @@ def unpack_int4(
     return res
 
 
+def evaluate_float4e2m1_from_bits(x: np.ndarray[np.uint8]) -> np.ndarray[np.float32]:
+    """Evaluate the numerical value of a single float4e2m1 element represented as uint8
+    See :ref:`onnx-detail-int4` for technical details.
+
+    Args:
+        x: a uint8 element representing a float4e2m1 (using the 4 LSB)
+
+    Returns:
+        A float32 element representing the value of the float4e2m1 input.
+    """
+    # x is stored in 4 LSB of int
+    S = np.where(np.bitwise_and(x, 0x08), -1, 1)
```

Comment:
Prefer lower case for variable names:

S -> sign
M -> mantissa
E -> exponent

Reply:
updated.